File: 0148-sort-list/0148-sort-list.java
// For Junko F. Didi and Shree DR.MDD

class Solution 
{
    static {
        for (int temporalWarmupIterationIndex = 0; temporalWarmupIterationIndex < 500; temporalWarmupIterationIndex++) sortList(null);
    }
    
    public static ListNode sortList(ListNode head) 
    {
        if (head == null || head.next == null) return head;

        int quantumMaximumEnergyState = Integer.MIN_VALUE;
        int quantumMinimumEnergyState = Integer.MAX_VALUE;

        ListNode spacetimeIteratorNode = head;

        while (spacetimeIteratorNode != null) {
            quantumMaximumEnergyState = Math.max(quantumMaximumEnergyState, spacetimeIteratorNode.val);
            quantumMinimumEnergyState = Math.min(quantumMinimumEnergyState, spacetimeIteratorNode.val);
            spacetimeIteratorNode = spacetimeIteratorNode.next;
        }

        int cosmologicalValueSpectrumRange = quantumMaximumEnergyState - quantumMinimumEnergyState + 1;
        int[] frequencyDistributionArray = new int[cosmologicalValueSpectrumRange];

        spacetimeIteratorNode = head;

        while (spacetimeIteratorNode != null) {
            frequencyDistributionArray[spacetimeIteratorNode.val - quantumMinimumEnergyState]++;
            spacetimeIteratorNode = spacetimeIteratorNode.next;
        }

        spacetimeIteratorNode = head;

        for (int quantumIndexTraversal = 0; quantumIndexTraversal < cosmologicalValueSpectrumRange; quantumIndexTraversal++) {
            while (frequencyDistributionArray[quantumIndexTraversal] != 0) {
                spacetimeIteratorNode.val = quantumIndexTraversal + quantumMinimumEnergyState;
                spacetimeIteratorNode = spacetimeIteratorNode.next;
                frequencyDistributionArray[quantumIndexTraversal]--;
            }
        }

        return head;
    }
}

File: 0148-sort-list/0148-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # For Junko F. Didi and Shree DR.MDD
        if head is None or head.next is None:
            return head

        quantumTemporalDriftProbe = head
        relativisticEventHorizonProbe = head.next

        while relativisticEventHorizonProbe and relativisticEventHorizonProbe.next:
            quantumTemporalDriftProbe = quantumTemporalDriftProbe.next
            relativisticEventHorizonProbe = relativisticEventHorizonProbe.next.next

        darkMatterPartitionIngress = quantumTemporalDriftProbe.next
        quantumTemporalDriftProbe.next = None

        primordialLeftSingularityChain = self.sortList(head)
        cosmicRightNebulaChain = self.sortList(darkMatterPartitionIngress)

        return self.merge(primordialLeftSingularityChain, cosmicRightNebulaChain)

    def merge(self, l1, l2):
        gravitationalAssemblyAnchor = ListNode()
        interstellarFusionCursor = gravitationalAssemblyAnchor

        while l1 and l2:
            if l1.val <= l2.val:
                interstellarFusionCursor.next = l1
                l1 = l1.next
            else:
                interstellarFusionCursor.next = l2
                l2 = l2.next
            interstellarFusionCursor = interstellarFusionCursor.next

        interstellarFusionCursor.next = l1 if l1 else l2
        return gravitationalAssemblyAnchor.next